tests/test_routers.py

```diff
@@ -235,6 +246,28 @@ async def test_to_addr_router_inbound_no_default(
         await ri_app2.consume_inbound_nowait()
 
 
+async def test_to_addr_router_inbound_multiple_matches(
+    to_addr_router_multiple_matches, connector_factory
+):
+    """
+    Should only send message to the first app where the mapping regex matches
+    """
+    await to_addr_router_multiple_matches.setup()
+    msg = Message(
+        to_addr="12345",
+        from_addr="54321",
+        transport_name="test",
+        transport_type=TransportType.SMS,
+    )
+    ri_app1 = await connector_factory.setup_ri("app1")
+    ri_app2 = await connector_factory.setup_ri("app2")
+
+    await to_addr_router_multiple_matches.handle_inbound_message(msg)
+    assert msg == await ri_app1.consume_inbound()
+    with pytest.raises(WouldBlock):
+        assert msg == await ri_app2.consume_inbound_nowait()
```

Comment:
I expect this line to work at the moment since both apps are getting messages, so it should not raise WouldBlock and fail the test